fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraBaseResource.java

```diff
@@ -94,4 +95,23 @@ protected void setUpJMSInfo(final UriInfo uriInfo, final HttpHeaders headers) {
         }
     }
 
+    /**
+     * Produce a baseURL for JMS events using the system property fcrepo.jms.baseUrl of the form http[s]://host[:port],
+     * if it exists.
+     *
+     * @return String the base Url
+     */
+    protected String getBaseUrlProperty() {
+        final String propBaseURL = System.getProperty("fcrepo.jms.baseUrl", "");
+        if (propBaseURL.length() > 0 && propBaseURL.indexOf(':') > -1) {
+            final String[] parts = propBaseURL.split(":");
+            if (parts.length == 3) {
+                return uriInfo.getBaseUriBuilder().clone().scheme(parts[0]).host(parts[1].substring(2)).port(
+                    Integer.valueOf(parts[2])).toString();
+            } else {
+                return uriInfo.getBaseUriBuilder().clone().scheme(parts[0]).host(parts[1].substring(2)).toString();
+            }
+        }
```

Comment:
IMHO, we should return the fcrepo.jms.baseUrl as-is without parsing and rebuilding.  The use case for this is users wanting to override the default generated baseURL, and I would expect it to be very frustrating if the value was altered in any way.